tests/roxctl/bats-tests/helpers.bash

```diff
@@ -9,10 +9,7 @@ fi
 load "${bats_helpers_root}/bats-support/load.bash"
 load "${bats_helpers_root}/bats-assert/load.bash"
 
-# luname outputs uname in lowercase
-luname() {
-  uname | tr '[:upper:]' '[:lower:]'
-}
+host_bin_platform="$(make --quiet host-bin-platform)"
```

Comment:
```suggestion
host_bin_platform="$(make --quiet --no-print-directory host-bin-platform)"
```
At least I've been tripped up by this a couple of time, I think, when there is an ancestor `make` process calling this script that runs in a separate working directory. Or is `--no-print-directory` only needed in conjunction with `-C`?

Reply:
I learned that `--quiet` is superset of `--no-print-directory`, at least with my version of `make`.
At least it works the same with `-C`:

```sh
$ cdrox
$ make -C operator image-flavor 
make: Entering directory '/home/misha/go/src/github.com/stackrox/stackrox/operator'
opensource
make: Leaving directory '/home/misha/go/src/github.com/stackrox/stackrox/operator'
$ make --quiet -C operator image-flavor
opensource
$ make --no-print-directory -C operator image-flavor
opensource
```

The difference between `--quiet` and `--no-print-directory` becomes more apparent if you use something like `make -C operator bundle-post-process`.
`--no-print-directory` suppresses messages about changing directory.
`--quiet` suppresses messages about changing directory plus echoing the command that's about to be ran.

Reply:
Hmm, as I said I've gotten tripped up by this in surprising ways. I experimented a bit, and it seems to me that (on GNU Make 4.3) the behavior is such that `--quiet` does only suppress printing directories if none of the ancestor `make`s printed directories, but not if an ancestor already printed directories - then you need `--no-print-directories`. We should never assume anything about how a caller invoked an ancestor `make`, so let's pass `--no-print-directory` in addition to `--quiet`.

```
$ cat Makefile 
.PHONY: bar bar-cd bar-npd bar-cd-npd

bar:
	$(MAKE) --quiet -C bar

bar-cd:
	cd bar && $(MAKE) --quiet

bar-npd:
	$(MAKE) --quiet --no-print-directory -C bar

bar-cd-npd:
	cd bar && $(MAKE) --quiet --no-print-directory

$ cat bar/Makefile
.PHONY: bar

bar:
	echo bar

$ make --quiet bar
bar # ok, expected
$ make --quiet -C . bar
bar #ok, expected
$ make bar
make --quiet -C bar # ok, expected due to omitting --quiet
bar
$ make -C . bar
make: Entering directory '/private/tmp/maketest' #ok, because of -C .. and no --quiet in first invocation
make --quiet -C bar
make[1]: Entering directory '/private/tmp/maketest/bar' # surprise! printed in spite of --quiet for this invocation
bar
make[1]: Leaving directory '/private/tmp/maketest/bar'
make: Leaving directory '/private/tmp/maketest'
$ make -C . bar-npd
make: Entering directory '/private/tmp/maketest' #ok, because of -C .. and no --quiet in first invocation
make --quiet --no-print-directory -C bar
bar #ok, no directories printing for second invocation
make: Leaving directory '/private/tmp/maketest'
```
The behavior of the `-cd` versions is the same as those using an explicit `-C`.

Because directories seem to be printed to stdout instead of stderr, this can break a `$(make --quiet -C .. tag)` expansion (as we use in the operator `Makefile`), but this breakage will _only_ be apparent when `make` on the `operator/` directory is invoked either with an explicit `-C` and no `--quiet`, or via `cd && make` from another `Makefile` that was invoked without `--quiet`.

Pretty complicated stuff, so I decided to just always pass both when using command expansion.

Reply:
Thanks, this got reproduced for me with `make` 4.2.1.

Makefiles calling scripts that call Makefiles which will start calling into other Makefiles some time later after refactoring.
If I was smarter I would come up with a comprehensive ruleset when `--quiet` or `--no-print-directory` is needed and when it can safely be omitted, but this can be a tiny problem that's not worth time spent.
I agree, let's do `--quiet --no-print-directory` together.